Runtime/src/index.ts

```diff
@@ -520,6 +629,28 @@ export class SwiftRuntime {
             swjs_release: (ref: ref) => {
                 this.heap.release(ref);
             },
+            swjs_sleep: (ms: number) => {
+                syncAwait(delay(ms));
+            },
+            swjs_sync_await: (
+                promiseRef: ref,
+                kind_ptr: pointer,
+                payload1_ptr: pointer,
+                payload2_ptr: pointer
+            ) => {
+                const promise: Promise<any> = this.heap.referenceHeap(promiseRef);
+                syncAwait(promise, kind_ptr, payload1_ptr, payload2_ptr);
+            },
+            swjs_sync_await_with_timeout: (
+                promiseRef: ref,
+                timeout: number,
+                kind_ptr: pointer,
+                payload1_ptr: pointer,
+                payload2_ptr: pointer
+            ) => {
+                const promise: Promise<any> = this.heap.referenceHeap(promiseRef);
+                syncAwait(promiseWithTimout(promise, timeout), kind_ptr, payload1_ptr, payload2_ptr);
+            },
```

Comment:
Can we expose only `swjs_sync_await` to minimize runtime functions? I think others can be implemented on Swift side.

Reply:
await_with_timeout definitely can be removed and easily implemented by callers. Not sure about the sleep one, as `setTimout` doesn't return a proper Promise implementing that from Swift side would be kind of ugly, and sleep is the most useful for imported c code.

Reply:
I've removed the withTimeout function. What do you think would be best regarding sleep?
1. Leave as is. Cons: Extra C function, extra JS function.
2. Have Swift construct the `delay` promise on first request with JSPromise-JSTimer code, cache it at `JSObject.global` for future use, then call it. Cons: Swift code will be ugly and less efficient, polluting global namespace.
3. Remove it from the Swift API and let users define their global JS promise if they want. Cons: While `sleep` is a special case of `wait`, it is arguably more useful and a surprise to users when they can't find it. And while adding the global JS function is trivial if you know about it, connecting these dots cold be difficult to newcomers. Plus, I hope actual C code can use it directly.
Or any other ideas?